src/test/java/org/jabref/logic/util/io/FileNameUniquenessTest.java

```diff
@@ -0,0 +1,56 @@
+package org.jabref.logic.util.io;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+public class FileNameUniquenessTest {
+
+    @TempDir
+    protected Path tempDir;
+
+    @Test
+    public void testGetNonOverWritingFileNameReturnsSameName(@TempDir Path tempDirectory) throws IOException {
+
+        assertFalse(
+                Files.exists(tempDirectory.resolve("sameFile.txt"))
+        );
+
+        String outputFileName = FileNameUniqueness.getNonOverWritingFileName(tempDirectory, "sameFile.txt");
+        assertEquals("sameFile.txt", outputFileName);
+
+    }
+
+    @Test
+    public void testGetNonOverWritingFileNameReturnsUniqueNameOver1Conflict() throws IOException {
+        Path dummyFilePath1 = tempDir.resolve("differentFile.txt");
+
+        Files.createFile(dummyFilePath1);
+
+        String outputFileName = FileNameUniqueness.getNonOverWritingFileName(tempDir, "differentFile.txt");
+        assertEquals("differentFile (1).txt", outputFileName);
+
+        Files.delete(dummyFilePath1);
```

Comment:
Deleting the file shouldn't be necessary as the `TempDir` should take care of this.